src/cache/cache.go

```diff
@@ -118,6 +129,23 @@ func (c *Cache) Prefetch(filename string) {
 	}
 }
 
+func (c *Cache) GetState(args *GetCacheStateArgs, reply *GetCacheStateReply) bool {
+    // TODO: Need some if statements around cache type here
+    c.mu.Lock()
+    defer c.mu.Unlock()
+    reply.State = c.chain
```

Comment:
This is bad. This will not copy by default, and will open us up to race issues. We need a MarkovChain.copy()

Reply:
(i.e. I believe this is a shallow copy)